Geometry/MTDCommonData/test/DD4hep_TestMTDIdealGeometry.cc

```diff
@@ -310,12 +310,12 @@ void DD4hep_TestMTDIdealGeometry::theBaseNumber(cms::DDFilteredView& fv) {
   thisN_.setSize(fv.navPos().size());
 
   for (uint ii = 0; ii < fv.navPos().size(); ii++) {
-    std::string name((fv.geoHistory()[ii])->GetName());
-    name.assign(name.erase(name.rfind('_')));
-    int copyN(fv.copyNos()[ii]);
-    thisN_.addLevel(name, copyN);
+    std::string_view name((fv.geoHistory()[ii])->GetName());
+    size_t ipos = name.rfind('_');
+    thisN_.addLevel((static_cast<std::string_view>((fv.geoHistory()[ii])->GetName())).substr(0, ipos),
```

Comment:
```suggestion
    thisN_.addLevel((name.substr(0, ipos),
```

Reply:
Two more cases of this line below should also be changed.

Reply:
given the nature of ```string_view``` it works even if ```name``` is local, I agree